Create new workspace creation helper and expose to python

DannyHindson opened the pull request:
**Description of work.**

Move a workspace creation helper method (create2DWorkspaceWithGeographicalDetectors) out of the unit test suite for the DiscusMultipleScatteringCorrection algorithm and into the WorkspaceCreationHelper so that it can be exposed to python. This will enable me to create a doc test in the algorithm's documentation page (which I'll do in a separate PR).

The create2DWorkspaceWithGeographicalDetectors method is useful for testing out the sparse instrument functionality that's in DiscusMultipleScatteringCorrection and MonteCarloAbsorption. If you set latitude or longitude=0, you can also use this to set up a ring of detectors at specific two theta values.

**To test:**

Run this script and then go to the Instrument Viewer or Show Detectors screen on the resulting test_ws workspace:

```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np
from testhelpers import WorkspaceCreationHelper

test_ws = WorkspaceCreationHelper.create2DWorkspaceWithGeographicalDetectors(1, 180, 1.0, 1)
ws_name = "test"
AnalysisDataService.addOrReplace(ws_name, test_ws)
```
There should be 180 detectors at two theta angles ranging from 0 to 179 degrees

Fixes #32195. This PR is one of two changes needed to fully address this issue

*This does not require release notes* because **it's only exposing a workspace helper function to python - mainly used by developers I think**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DannyHindson committed:
Expose new workspace helper function to python

Pull code to generate a workspace with detectors arranged at geographical
(ie lat and long angles) out into WorkspaceCreationHelper so that it can
be exposed to Python

DannyHindson committed:
Fix problem adding workspace to ADS

Fix problem adding a workspace created via create2DWorkspaceWithGeographicalDetectors
to the ADS

DannyHindson committed:
Few minor fixes

Allow instrument name to be specified so instrument viewer enabled
Fix Doxygen problems
Also clarify comment on when lat\long equals two theta

ConorMFinn commented on Framework/TestHelpers/src/InstrumentCreationHelper.cpp:
```diff
@@ -83,6 +83,46 @@ void addFullInstrumentToWorkspace(MatrixWorkspace &workspace, bool includeMonito
+  int i = 0;
+  constexpr double deg2rad = M_PI / 180.0;
+  const double angincRad = anginc * deg2rad;
+  auto R = 1.0;
```
Probably not overly relevant since it's just a move, but it seems like `R` could be `constexpr`?

ConorMFinn commented on Framework/TestHelpers/src/WorkspaceCreationHelper.cpp:
```diff
@@ -400,6 +400,23 @@ MatrixWorkspace_sptr create2DDetectorScanWorkspaceWithFullInstrument(int nhist,
+ */
+Workspace2D_sptr create2DWorkspaceWithGeographicalDetectors(const int nlat, const int nlong, const double anginc,
+                                                            const int nbins, const std::string &instrumentName) {
+  auto inputWorkspace = WorkspaceCreationHelper::create2DWorkspaceBinned(nlat * nlong, nbins, 0.5 /*x0*/);
```
What does this `/*x0*/` comment refer to?

DannyHindson replied:
It's the name of the parameter that receives the value 0.5 in the method Create2DWorkspaceBinned. I could never remember if this was the start x value or the bin width so I put this comment in. I can remove if it's a bit obscure!

DannyHindson replied:
I've declared a constexpr variable called x0 and removed the comment. Hopefully that's a bit more clear now

cailafinn reviewed (changes requested):
Looks good and works well, just a couple of minor comments:

DannyHindson committed:
Couple of small changes

Change definition of R to be constexpr
Replace comment hinting at parameter name with a variable

DannyHindson committed:
Change line number on cpp check suppression

cailafinn reviewed (approved):
Looks good to me now 👍🏻

DannyHindson commented:
@mantidproject/gatekeepers - please can someone merge this? It includes a change to the cpp check suppressions list that will stop the Jenkins cppcheck job failing
